fedimint-api/src/module/mod.rs

```diff
@@ -199,8 +199,11 @@ where
     }
 }
 
+/// Logic responsible for module's initialization, config generation and validation
+///
+/// Once the module config is ready, the module can be instantiated via `[Self::init]`.
 #[async_trait]
-pub trait FederationModuleConfigGen {
+pub trait ModuleInit {
```

Comment:
That name is less than ideal, but I don't have a better idea either rn. EDIT: maybe `ModuleMeta`?

Reply:
Module meta would mean "module about module" or something like that. I also find `ModuleInit` a bit weird.

`ModuleConstructor`?

Reply:
I like that

Reply:
You know what? Maybe I'll do bunch of renames in a follow up PR, because I'm afraid of conflict hell, as this got rather big.

Reply:
#1232